### ISSUE TITLE
Question: One modality is always separated

### ISSUE THREAD

**Issue by skarunan** [cantinilab/scconfluence#12]
Hi, 

My data set up is the following:
 
- scRNA (BD Rhapsody) with celltype annotation and scATAC (10x) for mice with no treatment
- scATAC (10x) for mice with treatment

My aim is to integrate these data sets and transfer the annotation labels from scRNA to the scATAC data sets, as discussed in #6 . 

While using your tool on these 3 data sets,  regardless of the `sinkhorn_loss = np.array([0.01, 0.06, 0.11, 0.16, 0.21, 0.26, 0.3 , 0.4 , 0.5 ])` and `reach = np.arange(0.5, 5, 0.5)` parameters, the scRNA data does not show integration on the UMAP plots. Below is a sample UMAP of how they look. I can upload a joint plot of the entire grid, but the outcome is only mildly different for the search space. However, when I just subset the RNA with one of the ATAC they are able to integrate. 

1. Would you suggest a finer grid search? 
2. Are there additional steps one need to care about while integrating more than 2 data sets? I basically added an additional encoder and added `cdist` between each of them. 

I would be happy to hear other suggestions you may have. Thanks for the nice tool.

Cheers

<img width="259" height="259" alt="RNA_separate" src="https://github.com/user-attachments/assets/2c268d04-9e36-4cb6-8342-f48f5331f9cd" />

<img width="667" height="407" alt="Subset" src="https://github.com/user-attachments/assets/96eddfb5-a3e1-4343-9313-5da46f1420cd" />

**Comment by skarunan** [comment]
Also, I am wondering what do you think of a work around, although it may take away one of the benefits of using scConfluence. First integrate e28-scRNA and e28-scATAC, then "transfer" the annotation of e28-scRNA to e28-scATAC. Then integrate e28-scATAC and e96-scATAC separately and annotate the e96-scATAC cells.

**Comment by jules-samaran** [comment]
Hi,
Thanks for reaching out!

I find the first UMAP you showed extremely surprising especially given the fact that the two modalities are completely mixed when you only keep one atac dataset.
Do you feed the two ATAC datasets as one anndata object with obs["batch"] indicating the dataset to which the ATAC cells belong to? 
Could you share with me few the lines of code you run to give the data to the model?

**Comment by skarunan** [comment]
Hi,

It was nice meeting you during the conference last month. The solution you proposed works. Many thanks for your help.

For the sake of thread completion: I had initially used the two atac datasets as two different anndata objects, as they are different experimental conditions. The solution you proposed works, (i.e. using one anndata object for the ATAC datasets with a batch variable for each ATAC data set). If I don't set the batch variable, then the ATAC data does not integrate well (see figures below). I explored a grid of sinkhorn loss and reach parameters of random 4k subset from each data set - to find a reasonable mixing of cells. 

It remains to be seen if the integration is biologically meaningful, after performing cell type annotations.

Subset - JointATAC - Without Batch:

<img width="3600" height="4400" alt="Image" src="https://github.com/user-attachments/assets/1b36d128-3d6f-4af6-96c0-c1912e9abb61" />

Subset - JointATAC - With Batch:

<img width="3600" height="4400" alt="Image" src="https://github.com/user-attachments/assets/c6813442-a93c-443a-a179-d344eec4d18e" />
